scalyr_agent/copying_manager.py

```diff
@@ -846,6 +856,13 @@ def run(self):
 
                             if result == "success":
                                 last_success = current_time
+
+                            # Rate limit based on amount of copied log bytes in a successful request
+                            if self.__rate_limiter:
+                                self.__rate_limiter.block_until_charge_succeeds(
```

Comment:
I also wonder if we should log a message in the rate limiter itself when we sleep when the limit is reached (with some limit on a maximum amount of log messages in some time period).

This would likely come handy in various troubleshooting scenario and answering the question of how many times we would actually exceeded the configured send rate and we need to wait / sleep.

And we could of course then also grep for that string in the corresponding end to end tests :)

Reply:
For the logging its planned as part of AGENT-397, a 5 minute log output if we rate limit. I'll probably have that in a separate PR.